src/models/connectionStore.ts

```diff
@@ -222,25 +261,62 @@ export class ConnectionStore {
     }
 
     /**
-     * Lookup credential store
+     * Lookup credential store with migration support
      * @param connectionCredentials Connection credentials of profile for password lookup
+     * @param isConnectionString Whether this is a connection string lookup
      */
     public async lookupPassword(
         connectionCredentials: IConnectionInfo,
         isConnectionString: boolean = false,
     ): Promise<string> {
-        const credentialId = ConnectionStore.formatCredentialId(
+        const profile = connectionCredentials as IConnectionProfile;
+        let savedCredential: Contracts.Credential;
+
+        // Generate credential ID using profile information (new format)
+        const credentialId = ConnectionStore.formatCredentialIdForCred(
+            profile,
+            CredentialsQuickPickItemType.Profile,
+        );
+
+        // First, try to get password from session storage for non-saved passwords
+        if (!profile.savePassword && ConnectionStore.shouldSavePasswordUntilRestart) {
+            const sessionPassword = this._sessionPasswords.get(credentialId);
+            if (sessionPassword) {
+                return sessionPassword;
+            }
+        }
+
+        // Read from credential store with new format id
```

Comment:
"If not found, read from..." (it's nice for comments to read like they're branches to logic instead of sequential steps when there's a likely early bailout).